examples/mxnet_mnist.py

```diff
@@ -0,0 +1,121 @@
+parser.add_argument('--batch-size', type=int, default=64,
+                    help='training batch size (default: 64)')
+parser.add_argument('--dtype', type=str, default='float32',
+                    help='training data type (default: float32)')
+parser.add_argument('--gpus', type=str, default='0',
+                    help='number of gpus to use (default: 0)')
+parser.add_argument('--epochs', type=int, default=10,
+                    help='number of training epochs (default: 10)')
+parser.add_argument('--lr', type=float, default=0.001,
+                    help='learning rate (default: 0.001)')
+parser.add_argument('--momentum', type=float, default=0.5,
+                    help='SGD momentum (default: 0.5)')
+args = parser.parse_args()
+
+logging.basicConfig(level=logging.INFO)
+logging.info(args)
+
+context = mx.cpu() if args.gpus is None or args.gpus == '0' \
+                   else mx.gpu(hvd.local_rank())
+
+# Step 2: data loading
+mx.random.seed(42)
+batch_size = args.batch_size
+input_shape = (1, 28, 28)
+
+train_iter = mx.io.MNISTIter(
+    image="data/train-images-idx3-ubyte",
+    label="data/train-labels-idx1-ubyte",
+    input_shape=input_shape,
+    batch_size=batch_size,
+    shuffle=True,
+    flat=False,
+    num_parts=hvd.size(),
+    part_index=hvd.rank()
+)
+
+val_iter = mx.io.MNISTIter(
+    image="data/t10k-images-idx3-ubyte",
+    label="data/t10k-labels-idx1-ubyte",
+    input_shape=input_shape,
+    batch_size=batch_size,
+    flat=False,
+    num_parts=hvd.size(),
+    part_index=hvd.rank()
+)
+
+
+# Step 3: define network
+def mlp():
+    # placeholder for data
+    data = mx.sym.var('data')
+    # Flatten the data from 4-D shape into 2-D 
+    # (batch_size, num_channel*width*height)
+    data = mx.sym.flatten(data=data)
+
+    # The first fully-connected layer and the corresponding activation function
+    fc1 = mx.sym.FullyConnected(data=data, num_hidden=128)
+    act1 = mx.sym.Activation(data=fc1, act_type="relu")
+
+    # The second fully-connected layer and the corresponding activation
+    # function
+    fc2 = mx.sym.FullyConnected(data=act1, num_hidden=64)
+    act2 = mx.sym.Activation(data=fc2, act_type="relu")
+
+    # MNIST has 10 classes
+    fc3 = mx.sym.FullyConnected(data=act2, num_hidden=10)
+
+    return fc3
+
+
+# Step 4: fit the model
+net = mlp()
+# Softmax with cross entropy loss
+loss = mx.sym.SoftmaxOutput(data=net, name='softmax')
+mlp_model = mx.mod.Module(symbol=loss, context=context)
+optimizer_params = {'learning_rate': args.lr * hvd.size()}
+opt = mx.optimizer.create('sgd', sym=net, **optimizer_params)
+opt = hvd.DistributedOptimizer(opt)
+
+initializer = mx.init.Xavier(rnd_type='gaussian', factor_type="in",
+                             magnitude=2)
+mlp_model.bind(data_shapes=train_iter.provide_data,
+               label_shapes=train_iter.provide_label)
+mlp_model.init_params(initializer)
+
+# Fetch and broadcast parameters
```

Comment:
Add `Horovod` to comment since it's Horovod specific.

Reply:
Done